### PR TITLE
Add support in HFSS for creating anisotropic impedance boundary condition and for the assignment of impedance boundaries to faces

### PR DESCRIPTION
**Pull request by OscarCapac**
Hello,

I've never contributed before to any git project so feel free to help if I'm doing things wrongly.

This contribution add the support in HFSS for:
- assigning an anisotropic impedance boundary to a sheet;
- assigning an impedance boundary to 1 or more faces;
- assigning an anisotropic impedance boundary to a single face.

The issue I created (#4248) for that was closed but pyaedt was still missing theses features from standard IronPython.

### CODE REVIEW

File: pyaedt/hfss.py [hunk 1]
```diff
@@ -3686,6 +3703,106 @@ def assign_impedance_to_sheet(self, sheet_name, sourcename=None, resistance=50,
...
+        ZyyResi : optional
+            Resistance value of yy component in ohms. The default is ``0``. If ``None``,
+            this parameter is disabled.
+        ZyyReac : optional
+            Reactance value of yy component in ohms. The default is ``0``. If ``None``,
+            this parameter is disabled.
+        is_infground : bool, optional
+            Whether the impedance is an infinite ground. The default is ``False``.
+        coord_sys : optional
+            The name of the coordinate system for the xy plane. The default is ``Global``.
```
Comment (PipKat) [inline pyaedt/hfss.py:3757]:
```suggestion
            Name of the coordinate system for the xy plane. The default is ``"Global"``.
```
We omit the article (The, An, A) at the beginning of the brief description. Also, a string default is enclosed in both backticks and quotation marks.)

File: pyaedt/hfss.py [hunk 2]
```diff
@@ -3686,6 +3703,106 @@ def assign_impedance_to_sheet(self, sheet_name, sourcename=None, resistance=50,
...
+            this parameter is disabled.
+        ZyyResi : optional
+            Resistance value of yy component in ohms. The default is ``0``. If ``None``,
+            this parameter is disabled.
+        ZyyReac : optional
+            Reactance value of yy component in ohms. The default is ``0``. If ``None``,
+            this parameter is disabled.
+        is_infground : bool, optional
+            Whether the impedance is an infinite ground. The default is ``False``.
+        coord_sys : optional
```
Comment (PipKat) [inline pyaedt/hfss.py:3756]:
```suggestion
        coord_sys : str, optional
```

File: pyaedt/hfss.py [hunk 3]
```diff
@@ -3036,6 +3046,8 @@ def create_boundary(
             Name of the boundary. The default is ``""``.
         is_infinite_gnd : bool, optional
             Whether the boundary is an infinite ground. The default is ``False``.
+        props : dict, optional
+            Additional properties for the boundary. The default is an empty dict ``{}``.
```
Comment (PipKat) [inline pyaedt/hfss.py:3050]:
```suggestion
            Additional properties for the boundary. The default is ``{}``.
```
Given that the data type is "dict," we assume that a developer knows that the default of ``{}`` is an empty dictionary.

File: pyaedt/hfss.py [hunk 4]
```diff
@@ -4651,6 +4768,171 @@ def assign_radiation_boundary_to_faces(self, faces_id, boundary_name=""):
...
+    @pyaedt_function_handler()
+    def assign_impedance_boundary_to_faces(
+        self, faces_id, boundary_name="", resistance=50, reactance=0, is_infground=False
+    ):
+        """Assign an impedance boundary to one or more faces.
+
+        Parameters
+        ----------
+        faces_id :
+            Face ID(s) to assign the impedance boundary condition to.
```
Comment (PipKat) [inline pyaedt/hfss.py:4780]:
```suggestion
            One or more face IDs to assign the impedance boundary condition to.
```
The data type needs to be added here and to a few other of the following parameters.

File: pyaedt/hfss.py [hunk 5]
```diff
@@ -4651,6 +4768,171 @@ def assign_radiation_boundary_to_faces(self, faces_id, boundary_name=""):
...
+        ZyyResi : optional
+            Resistance value of yy component in ohms. The default is ``0``. If ``None``,
+            this parameter is disabled.
+        ZyyReac : optional
+            Reactance value of yy component in ohms. The default is ``0``. If ``None``,
+            this parameter is disabled.
+        is_infground : bool, optional
+            Whether the impedance is an infinite ground. The default is ``False``.
+        coord_sys : optional
+            The name of the coordinate system for the xy plane. The default is ``Global``.
```
Comment (PipKat) [inline pyaedt/hfss.py:4885]:
```suggestion
            Resistance value of the xx component in ohms. The default is ``0``. If ``None``,
            this parameter is disabled.
        ZxxReac : optional
            Reactance value of the xx component in ohms. The default is ``0``. If ``None``,
            this parameter is disabled.
        ZxyResi : optional
            Resistance value of the xy component in ohms. The default is ``0``. If ``None``,
            this parameter is disabled.
        ZxyReac : optional
            Reactance value of the xy component in ohms. The default is ``0``. If ``None``,
            this parameter is disabled.
        ZyxResi : optional
            Resistance value of the yx component in ohms. The default is ``0``. If ``None``,
            this parameter is disabled.
        ZyxReac : optional
            Reactance value of the yx component in ohms. The default is ``0``. If ``None``,
            this parameter is disabled.
        ZyyResi : optional
            Resistance value of the yy component in ohms. The default is ``0``. If ``None``,
            this parameter is disabled.
        ZyyReac : optional
            Reactance value of the yy component in ohms. The default is ``0``. If ``None``,
            this parameter is disabled.
        is_infground : bool, optional
            Whether the impedance is an infinite ground. The default is ``False``.
        coord_sys : str, optional
            Name of the coordinate system for the xy plane. The default is ``"Global"``.
```

**Comment by PipKat** [comment]
@OscarCapac I'm the technical documentation specialist and did an editorial review of your PR. You did an AMAZING job at following our style!

**Comment by Samuelopez-ansys** [comment]
@OscarCapac Thank you for your contribution! I checked your implementation and it looks great, you just need one additional step, that is adding the unit test. All new methods must have unit test. Please check _unittest/test_20_HFSS.py, you will see how PyAEDT is testing all HFSS methods, for example, the method "create_perfecte_from_objects".

You can run the test locally, but first, you need to install tests dependencies. pip install .[tests]

Let us know if you are able to do it, if not we will help you.

Thank you again!

File: pyaedt/hfss.py [hunk 6]
```diff
@@ -3686,6 +3703,106 @@ def assign_impedance_to_sheet(self, sheet_name, sourcename=None, resistance=50,
...
+        sheet_name,
+        sourcename=None,
+        ZxxResi=0,
+        ZxxReac=0,
+        ZxyResi=0,
+        ZxyReac=0,
+        ZyxResi=0,
+        ZyxReac=0,
+        ZyyResi=0,
+        ZyyReac=0,
```
Comment (MaxJPRey) [inline pyaedt/hfss.py:3718]: I would suggest to follow pythonic naming convention if possible.
For instance: `z_xx_resistance`

**Pull request closed without merging**